src/lib.rs

```diff
@@ -98,19 +98,22 @@ impl Index {
     /// absolute index. negative indices are added to len.
     fn abs(&self, len: usize) -> Option<usize> {
         match self {
-            &Head(n) => Some(len.min(n)),
-            &Tail(n) => {
-                if len < n {
-                    None
-                } else {
-                    Some(len - n)
-                }
-            }
+            &Head(n) => ensure_within(0, n, len),
+            &Tail(n) => ensure_within(0, n, len).map(|n| len - n),
             Default => None,
         }
     }
 }
 
+/// Return Some(n) if min <= None < max, otherwise return None.
+fn ensure_within(min: usize, n: usize, max: usize) -> Option<usize> {
+    if n < min || max < n {
+        None
+    } else {
+        Some(n)
+    }
```

Comment:
Why not make this match the doc comment?
```suggestion
    if min <= n && n <= max {
        Some(n)
    } else {
        None
    }
```